src/python/pants/engine/goal.py

```diff
@@ -59,15 +60,17 @@ def Options(cls):
     # to get a reference to the outer class, which avoids implementers needing to subclass the inner
     # class in order to define their options values, while still allowing for the useful namespacing
     # of `Goal.Options`.
-    outer_cls = cls
+    outer_cls: Goal = cls
+
     class _Options(SubsystemClientMixin, Optionable, _GoalOptions):
       @classproperty
-      def options_scope(cls):
-        return outer_cls.name
+      def options_scope(cls) -> str:
+        name: str = outer_cls.name
```

Comment:
Is this typing still needed? The fix above to type `name` would do it you'd think.